Remove code coverage by directory view and backend API

Sush0512 opened the pull request:
Fixes #837.

marco-c commented on src/shipit_uplift/shipit_uplift/api.py:
```diff
@@ -21,7 +21,7 @@
 from shipit_uplift.config import SCOPES_USER, SCOPES_BOT, SCOPES_ADMIN
 from shipit_uplift import (
-    coverage_by_dir_impl, coverage_for_file_impl,
+     coverage_for_file_impl,
```
There's an additional whitespace here that shouldn't be there

marco-c commented on src/shipit_uplift/shipit_uplift/api.py:
```diff
@@ -383,8 +383,8 @@ def create_patch_status(bugzilla_id):
 
-def coverage_by_dir(path=''):
-    return coverage_by_dir_impl.generate(path)
+#def coverage_by_dir(path=''):
```
Let's remove it altogether, not just comment it out

marco-c commented on src/shipit_uplift/shipit_uplift/api.yml:
```diff
@@ -159,18 +159,7 @@ paths:
-      responses:
-        200:
-          description: Code coverage information by directory
+ 
```
Nit: Remove this additional newline

marco-c commented:
You can run the tests locally with `./please check shipit-uplift` and `./please check shipit-frontend`.

Sush0512 commented:
Hi @marco-c,
Please find the following updates
- removed extra spaces, lines
- synced the fork
- squashed the extra commits
- run the tests for shipit_uplift.

for shipit-frontend, it is throwing me an error saying there are no checks found for shipit-frontend.

marco-c commented:
> squashed the extra commits

There are several unrelated changes in the PR, can you try rebasing on master again?

Sush0512 commented:
@marco-c fixed it. Thank you :+1:

marco-c commented on src/shipit_frontend/src/App/CodeCoverage.elm:
```diff
@@ -1,249 +0,0 @@
-module App.CodeCoverage exposing (..)
```
There's probably some code under src/shipit_frontend that is referencing this that should be removed too.

marco-c commented:
There's still something that should be removed, see the build logs.

Have you tried running `./please run shipit-frontend`?

Sush0512 commented:
Hi @marco-c,
Sorry for the late reply. I had an issue with the run command and fixed it recently. 
I fixed the error causing snippet 
```
div [ class "collapse navbar-collapse" ]
        [ ul [ class "navbar-nav mr-auto " ]
            (viewNavDashboard model
                ++ [ li [ class "nav-item" ]
                        [ eventLink (ShowCodeCoverage Nothing) [ class "nav-link" ] [ text "Code Coverage" ]
                        ]
                   ]
            )
        , ul [ class "navbar-nav" ] (viewUser model)
]
```
by making the second parameter of `div` function empty, i.e, `div [ class "collapse navbar-collapse" ][]`. Is this fine? Now the build is successful.

marco-c commented on src/shipit_frontend/src/App.elm:
```diff
@@ -265,7 +234,7 @@ viewNavBar model =
         [ ul [ class "navbar-nav mr-auto " ]
```
We need to remove this "Code Coverage" ul completely.

marco-c commented on src/shipit_frontend/src/App.elm:
```diff
@@ -261,16 +230,7 @@ viewNavBar model =
-                        ]
-                   ]
-            )
-        , ul [ class "navbar-nav" ] (viewUser model)
```
Only the "Code Coverage" one should be removed, not also this ul.

marco-c commented on src/shipit_frontend/src/App.elm:
```diff
@@ -505,7 +455,8 @@ delta2url previous current =
 
                             _ ->
-                                [ "release-dashboard" ]
+                                [ "release-dashboard" ] 
```
Please remove unrelated changes here

marco-c commented on src/shipit_frontend/src/App.elm:
```diff
@@ -516,26 +467,12 @@ delta2url previous current =
             _ ->
                 Maybe.map
                     (Builder.prependToPath [])
-                    (Just builder)
```
Please remove unrelated changes here too

Sush0512 commented:
Carefully checked and updated all the files.
All the checks pass.

marco-c commented:
The build is still failing while validating the elm file.
You can run the check locally using `./please shell shipit-frontend` and then `elm-format-0.18 src/ --validate`.

Sush0512 commented:
Hi @marco-c 
I have run the commands you mentioned.
![shipit-frontend](https://user-images.githubusercontent.com/37044655/37640176-6fafc99e-2c3a-11e8-800b-4d421934a39b.png)
The build was successful but when tried to run the command `elm-format-0.18 src/ --validate.`  It throws an error saying please check the `elm-format-0.18` 
Can you please provide some insights on how to fix it, since there was no code added in the files.

garbas commented:
@Sush0512 we require the elm code to be formated with [elm-format](https://github.com/avh4/elm-format) tool. What you need to do is:

- enter the development environment, where this tool is available (`./please shell shipit-frontend`)
- format the failing file (`$ elm-format-0.18 src/App.elm`), you will have to press `y`
- commit the change

marco-c commented:
@Sush0512 need any help? Is what @garbas said clear?

Sush0512 commented:
Yes. Sorry for addressing late. Fixes #952

marco-c commented:
> Yes. Sorry for addressing late.

Don't worry, there's no rush! 😄

Sush0512 commented:
Thanks! :)
